python/src/convert.cpp

```diff
@@ -160,6 +160,10 @@ nb::ndarray<> mlx_to_dlpack(const array& a) {
 }
 
 nb::object to_scalar(array& a) {
+  if (a.size() != 1) {
+    throw std::invalid_argument(
+        "[convert] Only length-1 arrays can be converted to Python scalars.");
+  }
```

Comment:
I think the call to `.item` would already raise an error in that case so this is redundant. But maybe your intnetion was a nicer error message? (which I think it is).

Reply:
I didn't realize, but the error is a bit nicer so I guess it stays :)